core/frontend/src/Viewport.ts

```diff
@@ -1160,6 +1161,17 @@ export abstract class Viewport implements IDisposable {
     }
   }
 
+  /** This method enables or disables collection of timing statistics related to rendering frames.
+   * @param callback If defined, this enables collection of timing statistics related to rendering frames. If undefined, this disables the collection of timing statistics. When defined, this parameter specifies a callback function which receives the statistics every time a frame is rendered.
+   * @note The timing data collected using this method only collects the amount of time spent on the CPU. Due to performance considerations, time spent on the GPU is not collected. Therefore, these statistics are not a direct mapping to user experience.
+   * @note In order to avoid interfering with the rendering loop, take care to avoid performing any intensive tasks in your callback.
+   * @see [[FrameStats]]
+   * @alpha
+   */
+  public enableFrameStatsCallback(callback?: FrameStatsCallback): void {

```

Comment:
This allows at most one callback to be registered at any given time. Would an event be more appropriate?

Reply:
With your change now somebody (the app?) has to supply the event. That doesn't fix the problem of two or more callers fighting over who gets to monitor frame stats.
I had in mind that Viewport would have a `onFrameStats: BeEvent<(stats: FrameStats) => void`. Anybody could add or remove a listener for that event. Your FrameStatsCollector would record stats IFF at least one listener was registered.

Reply:
That makes sense.

Reply:
I changed the API to follow that pattern.